src/pages/workspaces/workspace/DeleteWorkspaceModal.ts

```diff
@@ -0,0 +1,167 @@
+import _ from 'lodash/fp';
+import pluralize from 'pluralize';
+import { useState } from 'react';
+import { b, div, h, label, p, span } from 'react-hyperscript-helpers';
+import { ButtonPrimary, Link, spinnerOverlay } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { TextInput } from 'src/components/input';
+import Modal from 'src/components/Modal';
+import { bucketBrowserUrl } from 'src/libs/auth';
+import colors from 'src/libs/colors';
+import { warningBoxStyle } from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+import { isGoogleWorkspace, WorkspaceWrapper as Workspace } from 'src/libs/workspace-utils';
+import { useDeleteWorkspaceState } from 'src/pages/workspaces/workspace/useDeleteWorkspaceState';
+
+interface DeleteWorkspaceModalProps {
+  workspace: Workspace;
+  onDismiss: () => void;
+  onSuccess: () => void;
+}
+
+const DeleteWorkspaceModal = (props: DeleteWorkspaceModalProps) => {
+  const [deleteConfirmation, setDeleteConfirmation] = useState('');
+  const { workspace, onDismiss, onSuccess } = props;
+  const {
+    workspaceResources,
+    loading,
+    deleting,
+    isDeleteDisabledFromResources,
+    workspaceBucketUsageInBytes,
+    collaboratorEmails,
+    hasApps,
+    deleteWorkspace,
+  } = useDeleteWorkspaceState({ workspace, onDismiss, onSuccess });
+
+  const getStorageDeletionMessage = () => {
+    return div({ style: { marginTop: '1rem' } }, [
+      'Deleting it will delete the associated ',
+      isGoogleWorkspace(props.workspace)
+        ? h(
+            Link,
+            {
+              ...Utils.newTabLinkProps,
+              href: bucketBrowserUrl(props.workspace.workspace.bucketName),
+            },
+            ['Google Cloud Bucket']
+          )
+        : 'Azure Storage Container',
+      ' and all its data',
+      workspaceBucketUsageInBytes !== undefined &&
+        span({ style: { fontWeight: 600 } }, [` (${Utils.formatBytes(workspaceBucketUsageInBytes)})`]),
+      '.',
+    ]);
+  };
+
+  const getResourceDeletionMessage = () => {
+    // On Azure, JupyterLab is returned as a runtime instead of an app (on GCP, Jupyter seems to be neither).
```

Comment:
I called this out to the IA team-- Jupyter for GCP is not included in the resources returned by `fetchWorkspaceResources` because that code is only getting Apps for GCP workspaces:
[https://github.com/DataBiosphere/terra-ui/blob/3af125e3c697c7457aea0b619c07dff7407[…]ee707/src/pages/workspaces/workspace/useDeleteWorkspaceState.ts](https://github.com/DataBiosphere/terra-ui/blob/3af125e3c697c7457aea0b619c07dff7407ee707/src/pages/workspaces/workspace/useDeleteWorkspaceState.ts#L56)

Reply:
Jupyter runs in a cloud environment (i.e.a VM) so I think this makes sense.

Reply:
I do wonder though if the error message (line 64 below) should not specifically call out RUNNING and ERROR as the the only states that are deletable-- for runtimes, I believe it is more lenient. I also don't know if we should be using some other term than "applications" given that it includes both apps and runtimes.

@mlilynolting please provide your opinion when you review!

Reply:
"Applications" is definitely misleading! Probably "cloud environments" would be more consistent.

Reply:
Keep in mind that the message is used for both GCP and Azure… if that makes any difference.

Reply:
@mlilynolting and do users think of WDS and Cromwell as "cloud environments"?

Reply:
Thinking about this more, since "runtimes" can pretty much always be deleted, if we are showing a message about things that cannot be deleted, they are almost certainly apps.

The bigger question is what we should show for things that can (and will) be deleted. Previous to this change, apparently we only warned the user that running applications will be deleted. Now we will also warn them if JupyterLab is running… which I would *think* we want to since the user would lose their persistent disk, right? @mlilynolting @admull what do you think? Do you agree that we should also warn if JupyterLab will be deleted, and if so, should we change the current message in some way? With the code as-is, this is what we show when deleting a workspace that has WDS and JupyterLab in it:

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/4092373f-ae9c-4581-863a-7aad322e012a)

Reply:
Perhaps change it to this?

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/5553e21b-c51a-409d-a4c7-5761c9ddfbab)

Reply:
@mlilynolting after discussing this with @admull and the workspaces team, we decided to simplify the messaging. Here is what I have pushed up, which I am also running by Kate.

You will get the message that deleting the workspace will delete cloud resources whenever the workspace does _not_ have an undeletable resources but _does_ have a deletable application (in Azure world, WDS or Cromwell) or a deletable runtime (in Azure world, JupyterLab, but does **not** include GCP Jupyter).

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/18bb6477-440b-4496-9820-c1c106d2563d)

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/4cd90860-ffe5-479f-ac73-6cfa3a664ab6)

Reply:
that looks great! We're also having a look to see if the runtimes list check can change to include GCP runtimes now.